src/profiler/profiler.h

```diff
@@ -1149,8 +1158,15 @@ struct ProfileOperator : public ProfileEvent {
     : ProfileEvent(name)
       , as_task_(name, &domain_)
       , name_(name)
-      , attributes_(attributes) {
-    SetCategories(domain_.name());
+      , attributes_(attributes)
+      , profiling_operator_(strcmp(name, "Dummy_Wait") && strcmp(name, "Custom"))
+      , profiling_task_(strcmp(name, "Dummy_Wait") && strcmp(name, "Custom")) {
+    if (strstr(name, "::")) {
```

Comment:
It would be great if you could turn this simple logic into a function with the explanation you have in this conversation, such as `bool IsSubCustomOperator(const char *)`. Otherwise, I am afraid it is going to be hard to comprehend for future developers.